PAPP-36658: generic action decorator for universal API calls (#98)
phantomcyber/splunk-soar-sdk: docs/api_reference.rst

.. autofunction:: soar_sdk.params.Param

Parameters for on poll
^^^^^^^^^^^^^^^^^^^^^^

> On poll functions require a specfic paramter class called OnPollParams. You should not override this class and use it as is.

Reviewer: ```suggestion
On poll functions require a specific parameter class called `OnPollParams`. You should use this class as-is, instead of overriding it.
```

Fixing spelling and making grammar a bit clearer

.. autoclass:: soar_sdk.params.OnPollParams

Parameters for generic action
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Generic action functions require a specfic paramter class called GenericActionParams. You should not override this class and use it as is.

Reviewer: ```suggestion
Generic action functions require a specific parameter class called `GenericActionParams`. You should use this class as-is, instead of overriding it.
```

Same as above

.. autofunction:: soar_sdk.action_results.OutputField

Generic Action Output
^^^^^^^^^^^^^^^^^^^^^

> For generic action functions we have provided a convinience class called GenericActionOutput. This class extends the ActionOutput class and adds a status_code and response_body field. You can use this class to return the response from the generic action.

Reviewer: typo: `convenience`

style: enclose class/field names in backticks for syntax highlighting
